app/processImage.js

```diff
@@ -78,24 +79,80 @@ function applySmartCrop(image, cropSize, callback) {
 }
 
 function applyCrop(image, cropSize, callback) {
-    sharp(image)
-        .resize(cropSize.width, cropSize.height)
-        .toBuffer(callback)
+    if (cropSize.widthPlus || cropSize.heightPlus) {
+        applySmartCrop(image, cropSize, callback)
+    } else {
+        if (cropSize.width < cropSize.originalWidth || cropSize.height < cropSize.originalHeight) {
+            applySmartCrop(image, cropSize, callback)
+        } else {
+            callback(null, image, {});
+        }
+    }
 }
 
 function applyCoverResize(image, cropSize, callback) {
-    sharp(image)
-        .resize(cropSize.width, cropSize.height)
-        .max()
-        .toBuffer(callback)
-}
 
-function rawImage(image, cropSize, callback) {
-    if (cropSize.width || cropSize.height) {
-        callback({}); // raise error if height or width is provided with rawType
-        return ;
+    if (cropSize.widthPlus || cropSize.heightPlus) {
+        sharp(image)
+            .resize(cropSize.width, cropSize.height)
+            .max()
+            .toBuffer(callback)
+    } else {
+        if (cropSize.width < cropSize.originalWidth || cropSize.height < cropSize.originalHeight) {
+            sharp(image)
+                .resize(cropSize.width, cropSize.height)
+                .max()
+                .toBuffer(callback)
+        } else {
+            callback(null, image, {});
+        }
     }
-    callback(null, image, {});
+}
+
+function applyBlurEffect(image, cropSize, callback) {
+    let blurImg = sharp(image);
+    let overlayImg = sharp(image);
+
+    blurImg
+        .metadata()
+        .then(function(metadata) {
+            if (cropSize.widthPlus && cropSize.heightPlus) {
```

Comment:
Can we add a documentation to what each each if/else is doing. Currently its confusing what we want to achieve in the conditions.